Mathlib/Topology/UnitInterval.lean

```diff
@@ -179,6 +193,87 @@ theorem two_mul_sub_one_mem_iff {t : ℝ} : 2 * t - 1 ∈ I ↔ t ∈ Set.Icc (1
 
 end unitInterval
 
+section partition
+
+namespace Set.Icc
+
+variable {α} [LinearOrderedAddCommGroup α] {a b c d : α} (h : a ≤ b) {δ : α}
+
+-- TODO: Set.projIci, Set.projIic
+/-- `Set.projIcc` is a contraction. -/
+lemma _root_.Set.abs_projIcc_sub_projIcc : (|projIcc a b h c - projIcc a b h d| : α) ≤ |c - d| := by
+  wlog hdc : d ≤ c generalizing c d
+  · rw [abs_sub_comm, abs_sub_comm c]; exact this (le_of_not_le hdc)
+  rw [abs_eq_self.2 (sub_nonneg.2 hdc), abs_eq_self.2 (sub_nonneg.2 <| monotone_projIcc h hdc)]
+  rw [← sub_nonneg] at hdc
+  refine (max_sub_max_le_max _ _ _ _).trans (max_le (by rwa [sub_self]) ?_)
+  refine ((le_abs_self _).trans <| abs_min_sub_min_le_max _ _ _ _).trans (max_le ?_ ?_)
+  · rwa [sub_self, abs_zero]
+  · exact (abs_eq_self.mpr hdc).le
+
+/-- Equally spaced points in a closed interval. -/
+def addNsmul (δ : α) (n : ℕ) : Icc a b := projIcc a b h (a + n • δ)
+
+lemma addNsmul_zero : addNsmul h δ 0 = a := by
+  rw [addNsmul, zero_smul, add_zero, projIcc_left]
+
+lemma addNsmul_eq_right [Archimedean α] (hδ : 0 < δ) :
+    ∃ m, ∀ n ≥ m, addNsmul h δ n = b := by
```

Comment:
I can't remember: do we prefer writing this with `tendsto`?

Reply:
It should be equivalent to `tendsto (addNsmul h δ) atTop (pure b)` but I think the spelling is too high-tech here ...

Reply:
I'm not familiar enough with the use of `Tendsto` to know whether this is too high-tech